python/lzreposync/src/lzreposync/__init__.py

```diff
@@ -1,15 +1,57 @@
 import argparse
+import logging
+import os.path
+from urllib.parse import urljoin
 
 from lzreposync import parse
 
 
 def main():
-    args_parser = argparse.ArgumentParser(description="Lazy reposync service")
-    args_parser.add_argument(
+    parser = argparse.ArgumentParser(
+        description="Lazy reposync service",
+        conflict_handler="resolve",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
         "--url",
         "-u",
         help="The target url of the remote repository of which we'll "
         "parse the metadata",
+        dest="url",
+        type=str,
+    )
+
+    parser.add_argument(
+        "-f",
+        "--file",
+        help="The target metadata file we want to parse",
+        dest="md_file",
+        type=str,
```

Comment:
You can find the primary file detection code in https://github.com/uyuni-project/obs-to-maven/blob/master/obs_maven/repo.py#L52-L59

Reply:
Done